src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.models.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturerStatement =
+                        connection.prepareStatement(insertManufacturerRequest,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer "
+                    + manufacturer + " to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getByIdRequest = "SELECT * FROM manufacturers "
+                + "where is_deleted = false AND id = ?;";
+        Manufacturer manufacturer = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getByIdManufacturerStatement =
+                        connection.prepareStatement(getByIdRequest)) {
+            getByIdManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getByIdManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                manufacturer = getManufacturer(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer from DB by id "
+                    + id, e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String readRequest = "SELECT * FROM manufacturers where is_deleted = false;";

```

Comment:
another possible variant of query (optional)
```suggestion
        String readRequest = "SELECT * FROM manufacturers where !is_deleted;";

```